internal/connector/connector.go

```diff
@@ -180,7 +175,7 @@ func Run(ctx context.Context, options Options) error {
 			Transport: httpTransportFromOptions(options.Server),
 		},
 		Headers: http.Header{
-			"Infra-Destination": {checkSum},
+			"Infra-Destination-Name": {options.Name},
```

Comment:
Could this cause existing connectors to be registered as a new instance if they didn't have a name before?

Ex:
1. Stand-up existing connector without a name, it gets a unique ID.
2. Upgrade the connector to specify a name (the same name as the unique ID).
3. Could this result in a duplicate destination?